### DISCUSSION TITLE
Still confused about router params is defineQuery

### DISCUSSION THREAD

**Discussion by marpe** [posva/pinia-colada#145] [Questions] [answered]
Sorry I closed the [discussion that was up earlier ](https://github.com/posva/pinia-colada/discussions/144)a little too soon. Since it's been marked as done, I'm opening a new one to continue. I made an example repo of the issue I'm having:

Actually I spoke too soon, can't get it to work even with these changes:

```ts
export const useDeck = defineQuery(() => {
  const route = useRoute("deck-deckId");
  // const deckId = computed(() => Number(route.params.deckId));

  const result = useQuery({
    key: () => ["decks", { deckId: Number(route.params.deckId) }],
    query: () => useRequestFetch()(`/api/decks`).then(decks => {
      return decks.find(d => d.id === Number(route.params.deckId));
    }) as Promise<(Deck & { cardCount: number }) | undefined>,
  });

  return {
    ...result
  }
});
```

Initial load works, but navigating back and forth doesn't trigger another fetch. Tried using `computed` as I've commented out, didn't work either.

Edit: I'll see if i can setup a minimum repro in a bit

Edit: [Repro here](https://stackblitz.com/~/github.com/marpe/pinia-colada-repro)

Sorry but I forgot to mention earlier that I'm using `compatibilityVersion: 4`

_Originally posted by @marpe in https://github.com/posva/pinia-colada/discussions/144#discussioncomment-11656169_

**Comment by posva** [comment]
This seems to come from Nuxt: ~~the ``useRequestFetch()(`/api/decks?deckId=${Number(route.params.deckId)}`)`` is only calling the API with the first call, only calling the API with one of the decks id. Subsequent calls fetch the exact same URL~~. For some reason the `useRoute()` returns a non reactive route. **This seems like a Nuxt bug**

**Reply by posva** [reply]
This seems to come from

- https://github.com/nuxt/nuxt/issues/28804#issuecomment-2327483434
- https://github.com/nuxt/nuxt/issues/20674#issuecomment-1534739499
- https://github.com/nuxt/nuxt/issues/21283#issuecomment-1569963217

One workaround I found is to use native vue router properties:

- replace `NuxtPage` with `RouterView`
- import `useRoute` from 'vue-router'

**Reply by posva** [reply]
I was wrong! This was a bug in defined queries: they were injecting local scope

**Reply by posva** [reply, marked as answer]
Fixed in 2307daf. I checked locally and it's working now 🎉

**Comment by wanderowski** [comment]
Hi everyone! Same for me. Any workarounds on this issue?
